component/prometheus/operator/podmonitors/crdmanager.go

```diff
@@ -0,0 +1,280 @@
+package podmonitors
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus"
+	"github.com/grafana/agent/component/prometheus/operator/podmonitors/config_gen"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/scrape"
+	toolscache "k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
```

Comment:
Can we give this a package alias that makes it clearer it's for Prometheus Operator? That was `v1.AddToScheme` later in this file is more clear about which scheme is being imported without having to scroll way back up.